### DISCUSSION TITLE
FDS file system limitation

### DISCUSSION THREAD

**Discussion by Clemeffect** [firemodels/fds#12005] [Q&A]
Hi all, 

I am running FDS simulation for my Master's dissertation. 
However, a problem occurs, usually around 60s of simulation. FDS stops and this error messages shows up:

_"forrtl: The requested operation could not be completed due to a file system limitation
forrtl: severe(38): error during write, unit 78, file C:\.... \electric room base_1_1.bf"_

I have run the simulations several times and it seems like it's always this file type having a problem. Though it feels random, because occasionally my simulation will run up the the 300s I have set as T_END. This issue really annoys me, and I have been stuck on it for quite some time now.  Space of my disk is not the issue.

Any help is welcomed. 

No idea if these could have an impact, but the simulation involves a fan, 6 sprinklers and a burner. (I can copy the fds code if necessary)

Thank you,
Clemeffect

**Comment by rmcdermo** [comment]
Boundary files can become huge.  You may want to limit the frequency of output by setting DT_BNDF.  Or limit the surfaces which get output by setting BNDF_DEFAULT=F on the MISC line and then turning on BNDF only for specific OBST.  See Sec. 22.5 Animated Boundary Quantities: The BNDF Namelist Group.

And, yes, we would have a better chance of diagnosing an FDS problem if you include your input file.  But if it is a system problem then we will likely get different results.

**Comment by drjfloyd** [comment]
A web search on the error indicates one possible cause is file fragmentation for very large files exceeding the limit for storing fragment locations on an NTFS drive espeically if there is also file compression enabled.  This could explain the randomness. Depending on what other files are being written by the operating system while FDS is running, the bf file could be experiencing differing levels of fragementation.

You could try defragmenting your drive before trying to run again. Randy's suggestion on reducing the fire size may also help.

**Comment by gforney** [comment]
this error message was written by the fortran run time library not FDS.  take another look at your disk.  the likely cause is a full disk. another cause could be permission problem but unlikely given that some output to this file has occurred.  upload your input file and I'll see if I can reproduce the problem

**Reply by Clemeffect** [reply]
I don't think it is due to a full disk since I am running 3 different FDS files (on 3 computers) on the same one drive disk, in different folders, and the other keep processing. The 3 files are really similar due, I am simulating scenarios. 

Here is my  code. It is probably not optimised since I'm completely knew to FDS.

&HEAD 	CHID='base extinction', TITLE='Base' /

&MESH 	IJK=114,74,34 XB=-0.2,11.2,-0.2,7.2,0,3.4 /

&TIME 	T_END=500 /

############################
########## Fire ############
############################

######## Reaction ########
&REAC ID = 'HEPTANE'
      FUEL               = 'N-HEPTANE',
      FYI                = 'Heptane, C_7 H_16',
      SOOT_YIELD         = 0.00 /

########## Ramp ###########
&RAMP ID='BURNER RAMP', T=0.0, F=0.0 /
&RAMP ID='BURNER RAMP', T=2.0, F=1.0 /
&RAMP ID='BURNER RAMP', T=15.0, F=1.0 /
&RAMP ID='BURNER RAMP', T=17.0, F=0.0 /


######## Surface ########
&SURF 	ID 		= 'BURNER',
	HRRPUA 		= 8000,
        COLOR 		= 'BLUE' /
        RAMP_Q		='BURNER RAMP'/ 
	
######## Fire ##########
&VENT XB = 2, 2.2, 0.4, 0.4, 0.6, 0.8, SURF_ID='BURNER' / 
	 
############################
######## Materials #########
############################

&MATL ID                   = 'FABRIC'
      HEAT_OF_COMBUSTION   = 17950.
      CONDUCTIVITY         = 0.2
      SPECIFIC_HEAT        = 0.9
      DENSITY              = 35. /

&MATL ID                   = 'GYPSUM_PLASTER', 
      CONDUCTIVITY         = 1,
      SPECIFIC_HEAT        = 0.84,
      DENSITY              = 175. /

&MATL ID                   ='STEEL'
      CONDUCTIVITY         = 0.045,
      SPECIFIC_HEAT        = 0.42,
      DENSITY              = 3400. /        

############################
######## Surfaces ##########
############################

&SURF ID 			= 'UPHOLSTERY', 
      COLOR 			= 'PURPLE', 
      MATL_ID 			= 'FABRIC',
      THICKNESS 		= 0.1, 
      BURN_AWAY 		= .TRUE. 
      IGNITION_TEMPERATURE	= 300.
      EXTINCTION_TEMPERATURE  	= 290.
      HRRPUA 			= 310 / 

&SURF ID 			= 'WALL', 
      RGB 			= 200,200,200, 
      MATL_ID 			= 'GYPSUM_PLASTER', 
      THICKNESS 		= 0.3 /

&SURF ID			= 'CABINET'
      COLOR 			= 'GRAY'
      MATL_ID			='STEEL'
      THICKNESS 		= 0.1 
      BURN_AWAY 		= .TRUE. 
      IGNITION_TEMPERATURE	= 1300.
      HRRPUA 			= 150 / #
      
############################
######## Geometry ##########
############################

######## Walls ##########
&OBST XB=-0.1,11.1,-0.1,0,0,3, 		SURF_ID = 'WALL' 	/ Wall 1
&OBST XB=-0.1,0,0,7,0,3, 		SURF_ID = 'WALL' 	/ Wall 2
&OBST XB=-0.1,11.1,7,7.1,0,3, 		SURF_ID = 'WALL' 	/ Wall 3
&OBST XB=11,11.1,0,7,0,3, 		SURF_ID = 'WALL' 	/ Wall 4
&OBST XB=-0.1,11.1,-0.1,7.1,3,3.1,	SURF_ID = 'WALL' 	/ Ceilling

######## Cabinets 1 ##########
&OBST XB = 1, 1.1, 0.2, 1.2, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 2.9, 3, 0.2, 1.2, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 1, 3, 0.2, 0.3, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 1, 3, 1.1, 1.2, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 1, 3, 0.2, 1.2, 0, 0.1, 		SURF_ID = 'CABINET' /
&OBST XB = 1, 3, 0.2, 1.2, 2.4, 2.5, 		SURF_ID = 'CABINET' /

&HOLE XB = 1.6, 2.4, 1.1, 1.2, 0.3, 0.4, /
&HOLE XB = 1.6, 2.4, 1.1, 1.2, 2.1, 2.2, /
&OBST XB = 1.2, 2.8, 0.4, 1.0, 0.2, 2.3, 	SURF_ID='UPHOLSTERY'/

######## Cabinets 2 ##########
&OBST XB = 3.0, 3.1, 0.2, 1.2, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 4.9, 5.0, 0.2, 1.2, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 3.0, 5.0, 0.2, 0.3, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 3.0, 5.0, 1.1, 1.2, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 3.0, 5.0, 0.2, 1.2, 0, 0.1, 		SURF_ID = 'CABINET' /
&OBST XB = 3.0, 5.0, 0.2, 1.2, 2.4, 2.5, 	SURF_ID = 'CABINET' /

&HOLE XB = 3.6, 4.4, 1.1, 1.2, 0.3, 0.4, /
&HOLE XB = 3.6, 4.4, 1.1, 1.2, 2.1, 2.2, /
&OBST XB = 3.2, 4.8, 0.4, 1.0, 0.2, 2.3, 	SURF_ID='UPHOLSTERY'/

######## Cabinets 3 ##########
&OBST XB = 5.0, 5.1, 0.2, 1.2, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 6.9, 7.0, 0.2, 1.2, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 5.0, 7.0, 0.2, 0.3, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 5.0, 7.0, 1.1, 1.2, 0, 2.5, 		SURF_ID = 'CABINET' /
&OBST XB = 5.0, 7.0, 0.2, 1.2, 0, 0.1, 		SURF_ID = 'CABINET' /
&OBST XB = 5.0, 7.0, 0.2, 1.2, 2.4, 2.5, 	SURF_ID = 'CABINET' /

&HOLE XB = 5.6, 6.4, 1.1, 1.2, 0.3, 0.4, /
&HOLE XB = 5.6, 6.4, 1.1, 1.2, 2.1, 2.2, /
&OBST XB = 5.2, 6.8, 0.4, 1.0, 0.2, 2.3, 	SURF_ID = 'UPHOLSTERY'/

######## Door ##########
&HOLE XB= 11,11.3,3,4,0,2.2 	/ Door
&VENT MB='ZMAX', SURF_ID='OPEN'	/

######## Fan ##########
&HOLE XB= 5,6,3,4,3.0,3.1, DEVC_ID='HEAT DETECTOR' /

&SURF ID			= 'FAN', 
      VEL			= 5.7, 
      COLOR			= 'RED' /

&OBST XB			= 5,6,3,4,3,3.3,	
      SURF_ID			= 'FAN', 
      DEVC_ID			= 'HEAT DETECTOR' /

&DEVC ID			= 'HEAT DETECTOR', 
      PROP_ID			= 'HEAT', 
      XYZ 			= 5.5,3.5,2.9 /

&PROP ID			= 'HEAT', 
      QUANTITY			= 'LINK TEMPERATURE', 
      RTI			= 3., 
      ACTIVATION_TEMPERATURE	= 40/


##########################
###### Sprinklers ########
##########################

###### General ######
&SPEC ID='WATER VAPOR' /

&PART ID			= 'my droplets', 
      DIAMETER			= 1000, 
      SPEC_ID			= 'WATER VAPOR' /

&PROP ID			= 'K-11', 
      OFFSET			= 0.04,
      PART_ID			= 'my droplets', 
      QUANTITY			= 'SPRINKLER LINK TEMPERATURE', 
      RTI			= 36., 
      C_FACTOR			= 0.4,
      ACTIVATION_TEMPERATURE	= 68., 
      OPERATING_PRESSURE	= 3.
      K_FACTOR			= 60
      PARTICLE_VELOCITY		= 3., 
      SPRAY_ANGLE		= 20.,80., 
      SMOKEVIEW_ID		= 'sprinkler_upright' 
      PRESSURE_RAMP		= 'PR'/

&RAMP ID='PR', T=1.0, F=4.0 /
&RAMP ID='PR', T=2.0, F=3.5 /
&RAMP ID='PR', T=3.0, F=3.0 /
&RAMP ID='PR', T=4.0, F=2.8 /


######## Devices ########
&DEVC ID='Spr-1', XYZ=2.7, 2.2, 2.9, PROP_ID='K-11', PIPE_INDEX=1 /
&DEVC ID='Spr-2', XYZ=5.5, 2.2, 2.9, PROP_ID='K-11', PIPE_INDEX=1 /
&DEVC ID='Spr-3', XYZ=8.2, 2.2, 2.9, PROP_ID='K-11', PIPE_INDEX=1 /
&DEVC ID='Spr-4', XYZ=2.7, 4.8, 2.9, PROP_ID='K-11', PIPE_INDEX=1 /
&DEVC ID='Spr-5', XYZ=5.5, 4.8, 2.9, PROP_ID='K-11', PIPE_INDEX=1 /
&DEVC ID='Spr-6', XYZ=8.2, 4.8, 2.9, PROP_ID='K-11', PIPE_INDEX=1 /

############################
######## Measures ##########
############################

&DEVC XYZ=2.7,3.5,0.5, 		QUANTITY='THERMOCOUPLE' , 	ID='TC1.1' /
&DEVC XYZ=2.7,3.5,1.0, 		QUANTITY='THERMOCOUPLE' , 	ID='TC1.2' /
&DEVC XYZ=2.7,3.5,1.5, 		QUANTITY='THERMOCOUPLE' , 	ID='TC1.3' /
&DEVC XYZ=2.7,3.5,2.0, 		QUANTITY='THERMOCOUPLE' , 	ID='TC1.4' /
&DEVC XYZ=2.7,3.5,2.5, 		QUANTITY='THERMOCOUPLE' , 	ID='TC1.5' /

&DEVC XYZ=5.5,3.5,0.5, 		QUANTITY='THERMOCOUPLE' , 	ID='TC2.1' /
&DEVC XYZ=5.5,3.5,1.0, 		QUANTITY='THERMOCOUPLE' , 	ID='TC2.2' /
&DEVC XYZ=5.5,3.5,1.5, 		QUANTITY='THERMOCOUPLE' , 	ID='TC2.3' /
&DEVC XYZ=5.5,3.5,2.0, 		QUANTITY='THERMOCOUPLE' , 	ID='TC2.4' /
&DEVC XYZ=5.5,3.5,2.5, 		QUANTITY='THERMOCOUPLE' , 	ID='TC2.5' /

&DEVC XYZ=8.2,3.5,0.5, 		QUANTITY='THERMOCOUPLE' , 	ID='TC3.1' /
&DEVC XYZ=8.2,3.5,1.0, 		QUANTITY='THERMOCOUPLE' , 	ID='TC3.2' /
&DEVC XYZ=8.2,3.5,1.5, 		QUANTITY='THERMOCOUPLE' , 	ID='TC3.3' /
&DEVC XYZ=8.2,3.5,2.0, 		QUANTITY='THERMOCOUPLE' , 	ID='TC3.4' /
&DEVC XYZ=8.2,3.5,2.5, 		QUANTITY='THERMOCOUPLE' , 	ID='TC3.5' /

&DEVC XYZ=2.0,0.4,2.3, 		QUANTITY='THERMOCOUPLE' , 	ID='TC-Cab1.1' /
&DEVC XYZ=2.0,1.0,2.3, 		QUANTITY='THERMOCOUPLE' , 	ID='TC-Cab1.2' /
&DEVC XYZ=4.0,0.4,2.3, 		QUANTITY='THERMOCOUPLE' , 	ID='TC-Cab2.1' /
&DEVC XYZ=4.0,1.0,2.3, 		QUANTITY='THERMOCOUPLE' , 	ID='TC-Cab2.2' /
&DEVC XYZ=6.0,0.4,2.3, 		QUANTITY='THERMOCOUPLE' , 	ID='TC-Cab3.1' /
&DEVC XYZ=6.0,1.0,2.3, 		QUANTITY='THERMOCOUPLE' , 	ID='TC-Cab3.2' /

&DEVC XB=5.0,6.0,3.0,4.0,3.05,3.05, 	QUANTITY='VOLUME FLOW'  , 	ID='Fan' /
&DEVC XB=11.1,11.1,3.0,4.0,0.0,1.1, 	QUANTITY='VOLUME FLOW'  ,	ID='Bottom Door' /
&DEVC XB=11.1,11.1,3.0,4.0,1.1,2.2,	QUANTITY='VOLUME FLOW'  ,	ID='Top Door' /

&SLCF PBX = 2.0, 		QUANTITY = 'TEMPERATURE' /
&SLCF PBY = 0.1,		QUANTITY = 'TEMPERATURE' /
&SLCF PBY = 1.2, 		QUANTITY = 'TEMPERATURE' /
&SLCF PBY = 1.3, 		QUANTITY = 'TEMPERATURE' /

&SLCF PBX = 5.5, 		QUANTITY='VELOCITY', VECTOR=.TRUE. /
&SLCF PBY = 3.5, 		QUANTITY='VELOCITY', VECTOR=.TRUE. /

&BNDF QUANTITY = 'WALL TEMPERATURE' / 
&BNDF QUANTITY = 'NET HEAT FLUX' / 
&BNDF QUANTITY = 'RADIATIVE HEAT FLUX' /

**Comment by gforney** [comment]
think fragmentation that jason pointed out is more likely than a full disk

**Comment by drjfloyd** [comment]
Don't use a onedrive disk. Use a local only local folder.

**Reply by Clemeffect** [reply]
Unfortunately, I can't... the local disks are too small

**Comment by drjfloyd** [comment]
Going to close this issue. Use of cloud based storage is risky. We can't control network interruptions, file locks that the cloud storage might impose, or other issues related to a remote file system. Consider getting a larger local disk. You can get a 2 Tb USB 3 SSD drive for ~ $100 USD.
